infrastructure/local-provisioners/azurerm_cosmosdb_collection.ts

```diff
@@ -112,53 +101,89 @@ const createCollectionIfNotExists = (
   });
 };
 
-export const run = async (config: IResourcesConfiguration) => {
-  const loginCreds = await login();
+interface IRunParams {
+  readonly resourceGroup: string;
+  readonly cosmosdbAccountName: string;
+  readonly cosmosdbDatabaseName: string;
+  readonly cosmosdbCollectionName: string;
+  readonly cosmosdbCollectionPartitionKey: string;
+}
+
+export const run = async (config: IRunParams) => {
+  const loginResult = await login();
 
   const client = new CosmosDBManagementClient(
-    (loginCreds as any).creds,
-    loginCreds.subscriptionId
+    loginResult.creds,
+    loginResult.subscriptionId
   );
 
   const databaseAccount = await client.databaseAccounts.get(
-    config.azurerm_resource_group,
-    config.azurerm_cosmosdb
+    config.resourceGroup,
+    config.cosmosdbAccountName
   );
 
   if (databaseAccount.documentEndpoint === undefined) {
     throw new Error("Cannot get databaseAccount.documentEndpoint");
   }
 
   const keys = await client.databaseAccounts.listKeys(
-    config.azurerm_resource_group,
-    config.azurerm_cosmosdb
+    config.resourceGroup,
+    config.cosmosdbAccountName
   );
 
   const dbClient = new DocumentClient(databaseAccount.documentEndpoint, {
     masterKey: keys.primaryMasterKey
   });
 
-  winston.info("Setup CosmosDB database");
+  winston.info(
+    `Making sure database exists: name=${config.cosmosdbDatabaseName}`
+  );
 
-  await createDatabaseIfNotExists(dbClient, config.azurerm_cosmosdb_documentdb);
+  await createDatabaseIfNotExists(dbClient, config.cosmosdbDatabaseName);
 
-  return Promise.all(
-    config.azurerm_cosmosdb_collections.map(
-      async collection =>
-        await createCollectionIfNotExists(
-          dbClient,
-          config.azurerm_cosmosdb_documentdb,
-          collection.name,
-          collection.partitionKey
-        )
-    )
+  winston.info(
+    `Making sure collection exists: name=${
+      config.cosmosdbCollectionName
+    } partitionKey=${config.cosmosdbCollectionPartitionKey}`
+  );
+  return createCollectionIfNotExists(
+    dbClient,
+    config.cosmosdbDatabaseName,
+    config.cosmosdbCollectionName,
+    config.cosmosdbCollectionPartitionKey
   );
 };
 
-checkEnvironment()
-  .then(() => readConfig(process.env.ENVIRONMENT))
-  .then(run)
-  .then(() =>
-    winston.info("Successfully deployed CosmosDB database and collections")
-  )
-  .catch((e: Error) => console.error(process.env.VERBOSE ? e : e.message));
+// check whether all required environment variables are set
+const missingEnvs = missingLoginEnvironment();
+if (missingEnvs.length > 0) {
+  console.error(`Missing required env vars: ${missingEnvs.join(", ")}`);
+  process.exit(-1);
+}
+
+const argv = yargs
```

Comment:
consider also commander which has typings for TS:

https://stackoverflow.com/questions/37199787/how-can-i-use-commander-npm-with-typescript

Reply:
what do you mean? it looks like `yargs`